### ISSUE TITLE
UnnecessaryAnnotationUseSiteTarget false positive with Kotlin 2.2

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#8212]
## Expected Behavior
Detekt aligns with Kotlin compiler in expectaitons.

## Observed Behavior
```
Example.kt:3:3: An annotation over a parameter doesn't need the use-site target @param. [UnnecessaryAnnotationUseSiteTarget]
```

## Steps to Reproduce

```kotlin
public class Example(
	public val path: String = "example.yml",
	@Language("yaml")
	public val content: String,
)
```
produces a kotlinc warning, so fixing that:
```kotlin
public class Example(
	public val path: String = "example.yml",
	@param:Language("yaml")
	public val content: String,
)
```
triggers Detekt.

## Context

```
> Task :ghlint-api:compileKotlin FAILED
e: warnings found and -Werror specified
w: file:///P:/projects/workspace/net.twisterrob.ghlint/modules/ghlint-api/src/main/kotlin/net/twisterrob/ghlint/rule/Issue.kt:22:2 This annotation is currently applied to the value parameter only, but in the
future it will also be applied to field.
- To opt in to applying to both value parameter and field, add '-Xannotation-default-target=param-property' to your compiler arguments.
- To keep applying to the value parameter only, use the '@param:' annotation target.
```

## Your Environment
* Version of detekt used: 1.23.8
* Version of Gradle used (if applicable): 9.0.0-rc1
* Version of Kotlin used: 2.2 (maybe 2.1 too)
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: Windows
* Link to your project (if it's a public repository):

**Comment by 3flex** [comment]
`Task :ghlint-api:compileKotlin FAILED`

Isn't this a compiler warning?

**Comment by TWiStErRob** [comment]
Warnings as errors is on on all my projects. See `w:` prefix.

**Comment by atulgpt** [comment]
Hi @TWiStErRob after the Kotlin version 2.2, they now target both param and property/field target. So fix should be
1. Before 2.2 version rule should be behave as it is as there is only one single target
2. On and after 2.2, when there are multiple targets are allowed then specifying the single target should not trigger the rule

Does the above sounds good to you?

Also we can optionally create a rule where if someone is adding all the targets by themself then we can trigger a warning stating that [`all`](https://kotlinlang.org/docs/annotations.html#defaults-when-no-use-site-targets-are-specified) can be used

**Comment by BraisGabin** [comment]
Shouldn't we just remove this rule?

- On 2.2 we don't care about it.
- On previous versions this rule will just make the migration to 2.2 more difficult.

**Comment by gobetti** [comment]
@BraisGabin I agree, looking at the [reasoning for introducing the rule](https://github.com/detekt/detekt/pull/2099), it seems that's reversed now with Kotlin 2.2
